diff --git a/compiler/Makefile.unix b/compiler/Makefile.unix
index 25beabc..73ae248 100755
--- a/compiler/Makefile.unix
+++ b/compiler/Makefile.unix
@@ -23,6 +23,7 @@ else
     ARCHFLAGS 	:= 
     LIBFLAGS	:= --mode=link --tag=CXX
     LLVM_CONFIG = $(word 1,$(wildcard /usr/bin/llvm-config*))
+    CXXFLAGS	+= -fPIC
     LIB_EXT     = so
 endif
 
@@ -72,6 +73,7 @@ ifeq ($(system), Darwin)
     dynamic : faust libfaust.$(LIB_EXT)
 else
     all : faust libfaust.a
+    dynamic : faust libfaust.$(LIB_EXT)
 endif
 
 ## On iOS, do not compile LLVM part
@@ -107,9 +109,13 @@ else
 	ranlib $@
 endif
 
+ifeq ($(system), Darwin)
 libfaust.$(LIB_EXT) : $(objects) libmain.o global.o
 	libtool -dynamic $(LIBFLAGS) $(objects) libmain.o global.o -lstdc++ `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version`  -macosx_version_min 10.6 -install_name /usr/local/lib/faust/libfaust.$(LIB_EXT) -o libfaust.$(LIB_EXT)
-
+else
+libfaust.$(LIB_EXT) : $(objects) libmain.o global.o
+	$(CXX) $(CXXFLAGS) -shared $(objects) libmain.o global.o -lstdc++ -Wl,-rpath,`$(LLVM_CONFIG) --libdir` `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version` -o libfaust.$(LIB_EXT)
+endif
 
 .PHONY: clean depend ctags parser
 
