--- compiler/generator/llvm-c-dsp.h~	2013-10-11 18:29:35.413581092 +0200
+++ compiler/generator/llvm-c-dsp.h	2013-10-11 18:45:04.051749097 +0200
@@ -40,11 +40,11 @@
  @{
  */
 
 	
-typedef struct llvm_dsp_factory {};
+typedef struct {} llvm_dsp_factory;
 
-typedef struct llvm_dsp {};
+typedef struct {} llvm_dsp;
 
 /**
  * Create a Faust DSP factory from a DSP source code. The source code is either given :
     - as a .dsp source filename (in which case the 'argc/argv' parameters have to be used)
