--- compiler/generator/llvm_dsp_aux.cpp~	2013-05-27 16:23:19.261033972 +0200
+++ compiler/generator/llvm_dsp_aux.cpp	2013-05-27 16:24:11.045077117 +0200
@@ -603,8 +603,12 @@
 }
 
 // Public C interface
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 EXPORT llvm_dsp_factory* createCDSPFactory(int argc, const char *argv[], 
                         const char* library_path, const char* draw_path, const char* name, 
                         const char* input, const char* target, 
                         char* error_msg, int opt_level)
@@ -699,5 +703,9 @@
 
 EXPORT void deleteCDSPInstance(llvm_dsp* dsp)
 {
     delete dsp; 
-}
\ Kein Zeilenumbruch am Dateiende.
+}
+
+#ifdef __cplusplus
+}
+#endif
--- compiler/generator/llvm_dsp_aux.hh~	2013-05-27 16:25:55.742975398 +0200
+++ compiler/generator/llvm_dsp_aux.hh	2013-05-27 16:32:05.586507770 +0200
@@ -251,8 +251,13 @@
 EXPORT void deleteDSPInstance(llvm_dsp* dsp);
 
 // Public C interface
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 EXPORT llvm_dsp_factory* createCDSPFactory(int argc, const char *argv[], 
                         const char* library_path, const char* draw_path, const char* name, 
                         const char* input, const char* target, 
                         char* error_msg, int opt_level);
@@ -292,8 +297,12 @@
 EXPORT llvm_dsp* createCDSPInstance(llvm_dsp_factory* factory);
 
 EXPORT void deleteCDSPInstance(llvm_dsp* dsp);
 
+#ifdef __cplusplus
+}
+#endif
+
 #ifdef WIN32
 
 #ifdef __cplusplus
 }
