commit 72772ee0b148ff0a1fad0e3f098a4913fbd1c2f8
Author: Marc Joliet <marcec@gmx.de>
Date:   Thu Nov 28 23:30:38 2013 +0100

    Support compiling libHTTPDfaust under Linux

diff --git a/architecture/httpdlib/src/Makefile b/architecture/httpdlib/src/Makefile
index 6dec360..d0edeea 100644
--- a/architecture/httpdlib/src/Makefile
+++ b/architecture/httpdlib/src/Makefile
@@ -10,27 +10,35 @@ system	?= $(shell uname -s)
 
 ifeq ($(system), Darwin)
 ARCHFLAGS :=  -arch i386 -arch x86_64
+LIB_EXT = dylib
 else
 ARCHFLAGS := 
+LIB_EXT = so
 endif
 
 CXXFLAGS ?= -O3 -Wall -Wuninitialized $(ARCHFLAGS)
 CXXFLAGS += `pkg-config --cflags libmicrohttpd` -I../../../architecture -I/usr/local/include -Wno-parentheses $(addprefix -I, $(subprojects)) -DINSTALL_PREFIX='"$(prefix)"'
 
-
-
+ifneq ($(system), Darwin)
+CXXFLAGS += -fPIC
+endif
 
 all : ../libHTTPDFaust.a
 
-dynamic : ../libHTTPDFaust.dylib
+dynamic : ../libHTTPDFaust.$(LIB_EXT)
 
 ../libHTTPDFaust.a : $(objects)
 	rm -f $@
 	ar cq $@ $(objects)
 	ranlib $@
 
-../libHTTPDFaust.dylib  :  $(objects)
+ifeq ($(system), Darwin)
+../libHTTPDFaust.$(LIB_EXT)  :  $(objects)
 	libtool -dynamic $(objects) -lc -lstdc++ `pkg-config --libs libmicrohttpd`  -macosx_version_min 10.6  -install_name /usr/local/lib/faust/libHTTPDFaust.dylib -o $@
+else
+../libHTTPDFaust.$(LIB_EXT)  :  $(objects)
+	$(CXX) $(CXXFLAGS) -shared $(objects) -lc -lstdc++ `pkg-config --libs libmicrohttpd` -o $@
+endif
 	
 depend :
 	makedepend -fMakefile -w120 -Y -- $(CXXFLAGS) -- $(sources)
