--- compiler/generator/llvm-c-dsp.h~	2013-05-28 16:55:45.783418427 +0200
+++ compiler/generator/llvm-c-dsp.h	2013-05-28 16:56:26.939813826 +0200
@@ -34,11 +34,11 @@
 #endif
 
 /* Opaque types */
 
-struct llvm_dsp_factory {};
+typedef struct {} llvm_dsp_factory;
 
-struct llvm_dsp {};
+typedef struct {} llvm_dsp;
 
 /**
  * Create a Faust DSP factory from a DSP source code. The source code is either given :
     - as a .dsp source filename (in which case the 'argc/argv' parameters have to be used)
--- architecture/faust/gui/CUI.h~	2013-05-28 16:55:45.530428290 +0200
+++ architecture/faust/gui/CUI.h	2013-05-28 16:57:56.254331089 +0200
@@ -91,9 +91,9 @@
 /***************************************
  *  Interface for the DSP object
  ***************************************/
 
-struct llvm_dsp_imp {};
+typedef struct {} llvm_dsp_imp;
 
 typedef llvm_dsp_imp* (* newDspFun) ();
 typedef void (* deleteDspFun) (llvm_dsp_imp* dsp);
 typedef int (* getNumInputsFun) (llvm_dsp_imp* dsp);
