diff --git a/compiler/Makefile.unix b/compiler/Makefile.unix
index 080d5dc..255bc9c 100755
--- a/compiler/Makefile.unix
+++ b/compiler/Makefile.unix
@@ -23,6 +23,7 @@ else
 ARCHFLAGS 	:= 
 LIBFLAGS	:= --mode=link --tag=CXX
 LLVM_CONFIG 	= $(word 1,$(wildcard /usr/bin/llvm-config*))
+CXXFLAGS	+= -fPIC
 LIB_EXT        = so
 endif
 
@@ -65,6 +66,7 @@ all : faust libfaust.a
 dynamic : faust libfaust.$(LIB_EXT)
 else
 all : faust libfaust.a
+dynamic : faust libfaust.$(LIB_EXT)
 endif
 
 libmain.o : libmain.cpp
@@ -91,9 +93,13 @@ libfaust.a : $(objects) libmain.o global.o
 	ranlib $@
 endif
 
+ifeq ($(system), Darwin)
 libfaust.$(LIB_EXT) : $(objects) libmain.o global.o
 	libtool -dynamic $(LIBFLAGS) $(objects) libmain.o global.o -lstdc++ `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version`  -macosx_version_min 10.6 -install_name /usr/local/lib/faust/libfaust.$(LIB_EXT) -o libfaust.$(LIB_EXT)
-
+else
+libfaust.$(LIB_EXT) : $(objects) libmain.o global.o
+	$(CXX) $(CXXFLAGS) -shared $(objects) libmain.o global.o -lstdc++ -Wl,-rpath,`$(LLVM_CONFIG) --libdir` `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version` -o libfaust.$(LIB_EXT)
+endif
 
 .PHONY: clean depend ctags parser
 
