--- Makefile~	2013-05-18 23:36:30.711815638 +0200
+++ Makefile	2013-05-18 23:37:20.795489642 +0200
@@ -76,9 +76,9 @@
 	mkdir -p $(prefix)/include/faust/
 	mkdir -p $(prefix)/lib/faust
 	install compiler/faust $(prefix)/bin/
 	install compiler/libfaust.a $(prefix)/lib/faust
-	[ -e compiler/libfaust.$(LIB_EXT) ] && install compiler/libfaust.$(LIB_EXT) $(prefix)/lib/faust
+	[ -e compiler/libfaust.$(LIB_EXT) ] && install compiler/libfaust.$(LIB_EXT) $(prefix)/lib/faust || echo libfaust.$(LIB_EXT) not available
 	cp compiler/libfaust.h  $(prefix)/include/faust/
 	cp compiler/generator/llvm-dsp.h  $(prefix)/include/faust/
 	# install architecture and faust library files
 	cp architecture/*.c $(prefix)/lib/faust/
