--- compiler/generator/llvm_dsp_aux.hh~	2013-05-27 17:25:58.066187882 +0200
+++ compiler/generator/llvm_dsp_aux.hh	2013-05-27 17:27:12.629916500 +0200
@@ -285,11 +285,11 @@
 EXPORT int getNumInputsCDSPInstance(llvm_dsp* dsp);
 
 EXPORT int getNumOutputsCDSPInstance(llvm_dsp* dsp);
 
-EXPORT void instanceInitCDSPInstance(llvm_dsp* dsp);
+EXPORT void instanceInitCDSPInstance(llvm_dsp* dsp, int samplingFreq);
 
-EXPORT void initCDSPInstance(llvm_dsp* dsp);
+EXPORT void initCDSPInstance(llvm_dsp* dsp, int samplingFreq);
 
 EXPORT void buildUserInterfaceCDSPInstance(llvm_dsp* dsp, UIGlue* interface);
 
 EXPORT void computeCDSPInstance(llvm_dsp* dsp, int count, FAUSTFLOAT** input, FAUSTFLOAT** output);
