--- compiler/main.cpp~	2013-06-02 14:48:21.361567558 +0200
+++ compiler/main.cpp	2013-06-02 14:48:59.112952756 +0200
@@ -31,19 +31,16 @@
     try {
         global::allocate();
         
         // Add time-out argument
-        int argc1 = argc + 2;
         const char* argv1[32];
-		assert(argc1 < 32);
+		assert(argc < 32);
         int i;
         for (i = 0; i < argc; i++) {
             argv1[i] = argv[i];
         }
-        argv1[i++] = "-t";
-        argv1[i++] = "120";
 
-        res = compile_faust_internal(argc1, argv1, "", "", 0, 0);
+        res = compile_faust_internal(argc, argv1, "", "", 0, 0);
     } catch (faustexception& e) {
         e.PrintMessage();
     }
     
