--- compiler/Makefile.unix.orig	2013-05-25 12:38:18.516398078 +0200
+++ compiler/Makefile.unix	2013-05-25 12:44:52.886173772 +0200
@@ -47,9 +47,9 @@
 ifeq ($(LLVM_VERSION), 3.2)
 LLVM_VERSION  = LLVM_32
 endif
 
-CXXFLAGS += -O3 -Wall -Wuninitialized 
+CXXFLAGS += -O3 -Wall -Wuninitialized -fPIC
 CXXFLAGS += -fvisibility=hidden -Wno-overloaded-virtual -Wno-parentheses $(addprefix -I, $(subprojects)) -DINSTALL_PREFIX='"$(prefix)"' 
 CXXFLAGS += -I`$(LLVM_CONFIG) --includedir` -I../architecture -D$(LLVM_VERSION) -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS
 CXXFLAGS += $(ARCHFLAGS)
 
@@ -74,12 +74,13 @@
 libfaust.a : $(objects) libmain.o global.o
 	libtool $(LIBFLAGS) $(objects) libmain.o global.o -static -o libfaust.a
 
 libfaust.$(LIB_EXT) : $(objects) libmain.o global.o
-	libtool -dynamic $(LIBFLAGS) $(objects) libmain.o global.o -lstdc++ `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version` -install_name /usr/local/lib/faust/libfaust.$(LIB_EXT) -o libfaust.$(LIB_EXT)
+	$(CXX) -shared $(objects) libmain.o global.o -lstdc++ -Wl,-rpath,`$(LLVM_CONFIG) --libdir` `$(LLVM_CONFIG) --ldflags` -lLLVM-`$(LLVM_CONFIG) --version` -o libfaust.$(LIB_EXT)
 
+libfaust: libfaust.a libfaust.$(LIB_EXT)
 
-.PHONY: clean depend ctags parser
+.PHONY: clean depend ctags parser libfaust
 
 parser :
 	bison -d -o parser/faustparser.cpp parser/faustparser.y
 	flex -I -oparser/faustlexer.cpp parser/faustlexer.l
