--- architecture/faust/audio/portaudio-dsp.h	2013-01-12 11:59:52.000000000 +0100
+++ portaudio-dsp.h	2013-01-25 10:50:29.294343738 +0100
@@ -160,9 +160,9 @@
         {
             if (pa_error(Pa_OpenDefaultStream(&fAudioStream, fDevNumInChans, fDevNumOutChans, paFloat32, fSampleRate, fBufferSize, audioCallback, this))) {
                 return false;
             }
-            if (pa_error(Pa_StartStream(fAudioStream)) {
+            if (pa_error(Pa_StartStream(fAudioStream))) {
                 return false;
             }
             return true;
         }
