--- a/src/s_audio_jack.c	2017-01-29 23:05:56.654971807 +0100
+++ b/src/s_audio_jack.c	2017-01-29 23:05:45.019932422 +0100
@@ -338,7 +338,7 @@
           sprintf(client_name,"pure_data_%d",client_iterator);
           client_iterator++;
           jack_client = jack_client_open (client_name, JackNoStartServer,
-            &status, NULL);
+            &status, "PD");
           if (status & JackServerFailed) {
             error("JACK: unable to connect to JACK server.  Is JACK running?");
             jack_client=NULL;
