--- a/pd/src/s_main.c	2013-01-23 12:26:48.000000000 +0100
+++ b/pd/src/s_main.c	2013-04-08 02:10:53.242707363 +0200
@@ -519,9 +519,9 @@
         strncat(buf, sbuf2, MAXPDSTRING-30);
         strncpy(sbuf2, buf, MAXPDSTRING-30);
     }
     sbuf[MAXPDSTRING-30] = 0;
-    strcat(sbuf, "/lib/pd-extended");
+    strcat(sbuf, "/lib/pd");
     if (stat(sbuf, &statbuf) >= 0)
     {
             /* complicated layout: lib dir is the one we just stat-ed above */
         sys_libdir = gensym(sbuf);
