--- Makefile.in	2009-12-11 10:37:10.000000000 +0100
+++ Makefile.in.b	2010-05-21 14:30:10.034536421 +0200
@@ -896,9 +896,9 @@
 
 
 skyeye_gui:
 	cd gui/qt4_src && rm -f skyeye_gui && make
-	cp -f gui/qt4_src/skyeye-gui $(prefix)/bin/
+	cp -f gui/qt4_src/skyeye-gui $(DESTDIR)$(prefix)/bin/
 lib:
 	cd third-party/libiberty && ./configure --prefix=$(prefix) --enable-shared ;make;
 	cd third-party/bfd/ && ./configure --prefix=$(prefix) --disable-werror --enable-shared --enable-targets=all;make;
 	cd third-party/opcodes/ && ./configure --prefix=$(prefix) --disable-werror --enable-shared --enable-targets=all;make;
@@ -919,10 +919,10 @@
 	cd third-party/bfd/ && make distclean
 	cd third-party/opcodes/ && make distclean
 	cd third-party/readline/ && make distclean
 install-data-hook:
-	rm -f -r $(prefix)/testsuite && mkdir $(prefix)/testsuite
-	cp -a $(top_srcdir)/testsuite/arm_hello $(prefix)/testsuite/arm_hello/
+	rm -f -r $(DESTDIR)$(prefix)/testsuite && mkdir $(DESTDIR)$(prefix)/testsuite
+	cp -a $(top_srcdir)/testsuite/arm_hello $(DESTDIR)$(prefix)/testsuite/arm_hello/
 #	rm -f -r $(prefix)/conf && mkdir $(prefix)/conf
 #	cp -a $(top_srcdir)/conf/* $(prefix)/conf
 
 # workaround for bfd library. we need to reconfigure it.
